### ISSUE TITLE
object index out of date after every update

### ISSUE THREAD

**Issue by Broxzier** [OpenRCT2/OpenRCT2#7834]
**OS:** Windows 10
**Version:** 0.2.0-234
**Commit/Build:** 5c2dd36

Lately starting the game after there has been an update takes much longer than before.

- [ ] Reproducible in RCT2 (vanilla)?
- [ ] Specific to multiplayer?

**Steps to reproduce:**
1. Download / build two versions of the game close to each other that haven't touched anything object-related in the code
2. Start up one of them, let it finish building the object database
3. Close it, and start the other build

Even while the objects are the same and nothing related to the object database has been changed in the code, you're greeted with the message `object index out of date` and have to wait for it to finish rebuilding. This seems unnecessary.

**Comment by AaronVanGeffen** [comment]
This appears to be happening on Linux, too.

**Comment by duncanspumpkin** [comment]
Is this still happening?

**Comment by Broxzier** [comment]
Yes, it's still happening.

**Comment by ZehMatt** [comment]
Perhaps we could add a verbose message that explains the reason for why it needs to rebuild the cache. I'm unable to do this on my end.

**Comment by Broxzier** [comment]
It's probably because the launcher downloads and unpacks the json objects for every new build. Then the game detects them as new files, even though they are the same. I'm not sure if the version is saved anywhere. We can assume the files remain untouched by the user, otherwise simply comparing the version number is not enough.

**Comment by Umdlye** [comment]
I don't wanna pressure anyone but just wanted to note that it would be quite helpful to not have this issue by the time the new save format is launched. People (like me) will probably want to test with lots of objects available and if the game is updated frequently around that time it could get pretty annoying, it takes my computer like 15+ minutes to finish my object index. ✌️

**Comment by IntelOrca** [comment]
The new save format is separate and doesn't affect this issue. If we don't check timestamps then we would have to check the content which will take longer to process. Anyway the issue is still availabel for anyone to fix.

**Comment by Umdlye** [comment]
Fair enough, I just figured that since object selection limits would be raised there would be more incentive to have more objects available which would make this issue more prominent.

**Comment by IntelOrca** [comment]
Hmm, I don't think the new save format will cause an explosion of new objects in the user's library but hmm maybe it might, I dunno.

**Comment by spacek531** [comment]
I don't get this issue, even when switching rapidly between a freshly-rebased debug build and a several-days-old release build. Is it still relevant?

**Comment by ZehMatt** [comment]
It still happens, will probably look into it this week.
